### PR TITLE
Added OAuth2 API and example for HiOrg-Server

### PR DESCRIPTION
**Pull request by MartinBoehmer**
HiOrg-Server supports humanitarian and civil protection organizations in Germany, e.g. the Red Cross.
I plan to implement their OAuth API in pac4j and CAS. So, scribejava is the point to start.

### CODE REVIEW

**Comment by MartinBoehmer** [comment]
Here is a backport for scribejava 5.3.0, which is compatible to PAC4J 3.1 and CAS 5.3
https://github.com/MartinBoehmer/scribejava/tree/hiorg-5.3.0

**Comment by MartinBoehmer** [comment]
Any plans on if and when this will be merged? The linked PRs for pac4j and CAS depend on this feature and its inclusion in a release of scribejava.

**Comment by kullfar** [comment]
Hi. I will try to look this PR today

**Comment by kullfar** [comment]
The code is ok. After I got credentials to run the example, PR will be merged.

**Comment by MartinBoehmer** [comment]
Great, thanks. I'll send you the details to run the example.

File: scribejava-apis/src/main/java/com/github/scribejava/apis/HiOrgServerApi20.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    @Override
+    protected String getAuthorizationBaseUrl() {
+        return "https://www.hiorg-server.de/api/oauth2/" + version + "/authorize.php";
+    }
+
+    @Override
+    public ClientAuthentication getClientAuthentication() {
+        return RequestBodyAuthenticationScheme.instance();
+    }
```
Comment (kullfar) [inline scribejava-apis/src/main/java/com/github/scribejava/apis/HiOrgServerApi20.java]: HiOrg-Server does work without this override.
And according to the https://tools.ietf.org/html/rfc6749#section-2.3.1
default is better.
"...the authorization server MAY support including the client credentials in the request-body... 
Including the client credentials in the request-body using the two
   parameters is NOT RECOMMENDED and SHOULD be limited to clients unable
   to directly utilize the HTTP Basic authentication scheme (or other
   password-based HTTP authentication schemes).
```

**Comment by kullfar** [comment]
I have removed getClientAuthentication override and have merged PR to the master.
Thanks for contributing!

**Comment by MartinBoehmer** [comment]
Great. Your're welcome. Thanks for maintaining this library.

**Pull request merged**